src/Chainweb/Miner/Core.hs

```diff
@@ -169,8 +169,10 @@ mine _ orig@(Nonce o) (TargetBytes tbytes) (HeaderBytes hbytes) = do
 -- hashed `BlockHeader`. If that layout changes, this functions need to be
 -- updated. The assumption allows us to iterate on new nonces quickly.
 --
+-- Recall: `Nonce` contains a `Word64`, and is thus 8 bytes long.
+--
 injectNonce :: Nonce -> Ptr Word8 -> IO ()
-injectNonce (Nonce n) buf = poke (castPtr buf) n
+injectNonce (Nonce n) buf = pokeByteOff (castPtr buf) 278 n
```

Comment:
Seems to be the correct magic number.

Reply:
Maybe adding a comment with a reference to https://github.com/kadena-io/chainweb-node/wiki/Block-Header-Binary-Encoding ?